### ISSUE TITLE
[BUG] SetFilters function in useTable hook not sync with url properly

### ISSUE THREAD

**Issue by mustafaulukaya** [pankod/refine#2432]
**Describe the bug**
setFilters function in useTable hook not working properly. When you set filters like this

```
let filters: CrudFilters = [];
setFilters(filters, "replace");
```

The page URL did not change. In app.tsx, I pass the "true" parameter via syncWithLocation.

**To Reproduce**
1. Go to 'App.tsx' and pass "true" value to "syncWithLocation" parameter.
2. Then use 'useTable' hook and get 'setFilters' function.
3. First use 'setFilters' function to real filter. Then call it with empty array.

**Expected behavior**
The page URL changes with filters.

**Screenshots**
![image](https://user-images.githubusercontent.com/24412394/188372848-5702222e-cff9-4f5d-9941-509f76b6c6e1.png)

**Videos**
https://user-images.githubusercontent.com/24412394/188372981-037b92a7-ffa0-4ce7-966c-4cffdf108197.mp4

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: chrome
 - Refine Version: 3.56.11
 - Used dataProvider: custom

**Comment by omeraplak** [comment]
Hey @mustafaulukaya ,
We've released [`@pankod/refine-core@3.65.3`](https://github.com/pankod/refine/releases/tag/%40pankod%2Frefine-core%403.65.3). Could you approve this fix?

**Comment by mustafaulukaya** [comment]
Hey,

The bug was successfully fixed. Thank you and your team. Have a great day!
